homeassistant/components/switch_as_x/light.py

```diff
@@ -0,0 +1,32 @@
+"""Light support for switch entities."""
+from __future__ import annotations
+
+from homeassistant.components.switch.light import LightSwitch
```

Comment:
We're not allowed to directly depend on details of another platform. If we want to allow other integrations to use the entity class we need to export it like this:
https://github.com/home-assistant/core/blob/36385396b09a14e3f179f74454cb883107dd4564/homeassistant/components/mjpeg/__init__.py#L11-L16

Reply:
Isn't this integration replacing the LightSwitch? As in, shouldn't it just implement its own platform entity instead?

Reply:
Changed as suggested by @MartinHjelmare

Reply:
Any comments on Franck's question?

Reply:
The class definition is *moved* to switch_as_x/light.py in commit `8e6dc52`, I'd prefer to not duplicate the code unless there's a good reason to do that?